### ISSUE TITLE
Remove malloc.h for clang

### ISSUE THREAD

**Issue by songlinm** [Refinitiv/Real-Time-SDK#82]
This error is found when compiling with clang:

```
Elektron-SDK/Cpp-C/Eta/Impl/Codec/dataDictionary.c:17:10: fatal error: 'malloc.h' file not found
#include <malloc.h>
         ^~~~~~~~~~
1 error generated.
make[2]: *** [Cpp-C/Eta/Impl/Codec/CMakeFiles/librssl.dir/dataDictionary.c.o] Error 1
make[1]: *** [Cpp-C/Eta/Impl/Codec/CMakeFiles/librssl.dir/all] Error 2
make: *** [all] Error 2

```
Apparently, malloc.h is deprecated and should be replaced with stdlib.h, see this [post](https://stackoverflow.com/a/12973361). Not sure if the header is there for Windows compilation, if so, can we protect it with a macro `# ifndef __clang__`, similar to the fix for [this issue](https://github.com/open62541/open62541/issues/834)?

**Comment by bberner** [comment]
Thank you for your feedback. A JIRA issue has been created to change malloc.h include to stdlib.h include. This will be done in a future release.

**Comment by vlevendel** [comment]
Fix is included with tag, Real-Time-SDK-2.0.1.L1.
